### ISSUE TITLE
JSON array not updating items correctly

### ISSUE THREAD

**Issue by Fyroman** [nuxt/content#582]
First problem is in database file:

```
if (document) {
  this.items.update({ $loki: document.$loki, meta: document.meta, ...item })
  return // this here
}
```

It only updates the first item and breaks execution.

Second problem is that there is no deletion... For quick workaround I changed the upper "return" to "continue", but it would then only update existing items and add new ones, but if any were removed from file it wouldn't notice them and remove them from database.

Edit: Yeah, alright, adding this.removeFile(path) to the beginning of updateFile method kinda does the job, but I guess it breaks everything else.

**Comment by benjamincanac** [comment]
@Fyroman Do you mean during hot reload?

**Comment by Fyroman** [comment]
> @Fyroman Do you mean during hot reload?

No, when overwriting file with writeFileSync. It's a 430mb file with 10 000 items (separate files crash), so using PUT route doesn't work well, crashes server instance.

In fact, it's possible PUT route doesn't remove leftover items as well, but I can't say for sure, I didn't test it too much because of large data size.

**Comment by Tahul** [comment]
Hello;

I am labelling this as both v1 & v2 because your use case is interesting for further development.

Could you provide more informations about that 430mb file?

I am really interesting in knowing how you are using such file for a web project, and how @nuxt/content could serve that purpose!
